### PR TITLE
Fix/sign in with google

### PR DESCRIPTION
**Pull request by rBergCode**
# Summary
When creating an account or signing in with google, if the profile was already created the app would not move the map screen.

# Details
While fixing this, I also remove the `addProfile` function as it was useless in our case

### CODE REVIEW

File: app/src/main/java/ch/hikemate/app/model/profile/ProfileRepositoryFirestore.kt [hunk 1]
```diff
@@ -39,17 +38,29 @@ class ProfileRepositoryFirestore(private val db: FirebaseFirestore) : ProfileRep
...
       onSuccess: (Profile) -> Unit,
-      onFailure: (Exception) -> Unit
+      onFailure: (Exception) -> Unit,
+      context: Context
   ) {
     if (fireUser == null) {
       onFailure(Exception("User is null"))
       return
     }
-    val context = ApplicationProvider.getApplicationContext<Context>()
```
Comment (Wylwi) [inline app/src/main/java/ch/hikemate/app/model/profile/ProfileRepositoryFirestore.kt:48]: Question: Why is this line removed? What was the issue with it?

**Review by Wylwi — COMMENTED** [review]
Just a question about a line! 

I also have a more general question, why do you say the addProfile is useless? Can't it be useful later on?

File: app/src/main/java/ch/hikemate/app/model/profile/ProfileRepositoryFirestore.kt (hunk 1 above)
Comment (Wylwi) [reply app/src/main/java/ch/hikemate/app/model/profile/ProfileRepositoryFirestore.kt:48]: Oh okay thanks!

**Comment by rBergCode** [comment]
Also, `addProfile` was useless as we now have a `createProfile` function

**Review by Wylwi — APPROVED** [review]
Okay, thanks for answering my questions! Looks like a clean refactor! Try to have a better PR and branch name next time 😉

**Comment by rBergCode** [comment]
We will ignore sonarcloud report for this PR as we removes tests but the coverage is still good
